domain/schema/model/sql/0008-space.sql

```diff
@@ -3,22 +3,18 @@ CREATE TABLE space (
     name TEXT NOT NULL
 );
 
-CREATE UNIQUE INDEX idx_spaces_uuid_name
-ON space (name);
+CREATE UNIQUE INDEX idx_spaces_uuid_name ON space (name);
 
 CREATE TABLE provider_space (
     provider_id TEXT NOT NULL PRIMARY KEY,
     space_uuid TEXT NOT NULL,
-    CONSTRAINT fk_provider_space_space_uuid
-    FOREIGN KEY (space_uuid)
-    REFERENCES space (uuid)
+    CONSTRAINT fk_provider_space_space_uuid FOREIGN KEY (space_uuid) REFERENCES space (uuid)
 );
 
-CREATE UNIQUE INDEX idx_provider_space_space_uuid
-ON provider_space (space_uuid);
+CREATE UNIQUE INDEX idx_provider_space_space_uuid ON provider_space (space_uuid);
 
 INSERT INTO space VALUES
-(0, 'alpha');
+('019593ec-84f2-7772-bad2-7a770aed04bc', 'alpha');
```

Comment:
oof `core` is now implicitly coupled with `domain`

Reply:
At least a comment, either here or there, is required to explain this connection

Reply:
Normally we create a domain to handle these cases...

See life as an example:

https://github.com/juju/juju/blob/393517620f556415d2d7aae2c56b6d0f9fc47e6f/domain/life/life.go#L8-L30

Reply:
Fair point. I think a random uuid is a more egregious case than the number 0 though. 

Arguably though highlighting this coupling with something extreme could be a good idea

Reply:
> Normally we create a domain to handle these cases 

Indeed I agree, but again this patch does not address this. The `core.AlphaSpaceID` const is used in many packages where we don't want to bring a domain dependency (namely state/). I'm positive we can get rid of this later (for example when endpoint bindings are fully migrated to dqlite), and it's a bigger refactoring.